common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -0,0 +1,279 @@
+    /**
+     * The version of the calling library.
+     */
+    private String mLibraryVersion;
+
+    /**
+     * Constructor of MicrosoftAuthorizationRequest.
+     */
+    public MicrosoftAuthorizationRequest(final String responseType,
+                                         @NonNull final String clientId,
+                                         @NonNull final String redirectUri,
+                                         final String state,
+                                         final Set<String> scope,
+                                         @NonNull final URL authority,
+                                         @NonNull final String authorizationEndpoint,
+                                         final String loginHint,
+                                         final UUID correlationId,
+                                         final PkceChallenge pkceChallenge,
+                                         final String extraQueryParam,
+                                         final String libraryVersion) {
+        super(responseType, clientId, redirectUri, state, scope);
+
+        if (StringUtil.isEmpty(redirectUri)) {
+            throw new IllegalArgumentException("redirect Uri is empty");
+        }
+
+        if (StringUtil.isEmpty(authorizationEndpoint)) {
+            throw new IllegalArgumentException("Authorization endpoint is empty");
+        }
+
+        mAuthority = authority;
+        mAuthorizationEndpoint = authorizationEndpoint;
+        mLoginHint = loginHint;
+        mCorrelationId = correlationId;
+        mPkceChallenge = pkceChallenge;
+        mExtraQueryParam = extraQueryParam;
+        mLibraryVersion = libraryVersion;
+    }
+
+    /**
+     * Default constructor of {@link MicrosoftAuthorizationRequest}
+     */
+    public MicrosoftAuthorizationRequest() {
+        super();
+    }
+
+    public URL getAuthority() {
+        return mAuthority;
+    }
+
+    public void setAuthority(final URL authority) {
+        mAuthority = authority;
+    }
+
+    public String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    public void setAuthorizationEndpoint(final String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public String getLoginHint() {
+        return mLoginHint;
+    }
+
+    public void setLoginHint(final String loginHint) {
+        mLoginHint = loginHint;
+    }
+
+    public UUID getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public void setCorrelationId(final UUID correlationId) {
+        mCorrelationId = correlationId;
+    }
+
+    public PkceChallenge getPkceChallenge() {
+        return mPkceChallenge;
+    }
+
+    public void setPkceChallenge(final PkceChallenge pkceChallenge) {
+        mPkceChallenge = pkceChallenge;
+    }
+
+    public String getExtraQueryParam() {
+        return mExtraQueryParam;
+    }
+
+    public void setExtraQueryParam(final String extraQueryParam) {
```

Comment:
No notes about the expected format....does this need to be encoded? Can I set multiple query params at once? Or do I call this in a loop?